src/container/cli_internal.h

```diff
@@ -53,8 +76,22 @@ dc_cont2hdl(struct dc_cont *dc, daos_handle_t *hdl)
 	daos_hhash_link_key(&dc->dc_hlink, &hdl->cookie);
 }
 
+void
+dc_cont_hdl_unlink(struct dc_cont *dc)
+{
+	int i;
+
+	daos_hhash_link_delete(&dc->dc_hlink);
+	for (i = 0; i < DC_CONT_CACHE_NR; i++) {
+		if (dc == dc_cont_local[i]) {
+			dc_cont_local[i] = NULL;
+			dc_cont_cookie[i] = 0;
```

Comment:
looks a potential problem, if unlink by thread A, but it still in the local cache at thread B, that could cause thread B get invalid dc_cont?

Reply:
Indeed, trying to fix this by holding extra references for cache, and it will be put when daos_fini() is called.